lizard.py

```diff
@@ -759,8 +766,14 @@ def get_warnings(code_infos, option):
 
 def print_result(result, option, scheme):
     result = print_and_save_modules(result, option.extensions, scheme)
+    warning_count = 0
+    warning_nloc = 0
     warnings = get_warnings(result, option)
-    warning_count, warning_nloc = print_warnings(option, scheme, warnings)
+    if list(warnings):
+        warnings = get_warnings(result, option)
```

Comment:
```
 +    if list(warnings):
 ```
Yes, this line will exhaust the generator. But moving it to get_warnings isn't a good idea since the "warning_only" option still need the result to be a generator.

@rakhimov sorry for the lack of comments for `get_warnings`. It obviously doesn't explain itself very well. What would be a better name?

@mdmedley have you tried adding any unit test for the change? There hasn't been enough UT for the UI. So it will be great if you can add some. Otherwise, I can do it too. Thanks for your contribution!